.github/workflows/release-notes.yaml

```diff
@@ -0,0 +1,85 @@
+name: Release Notes
+
+on:
+  pull_request:
+    types: [ assigned, opened, synchronize, reopened, edited ]
+env:
+  CURRENT_RELEASE_JSON_FILE_PATH: tool/release_notes/release_notes.json
+jobs:
+  release-preparedness:
+    runs-on: ubuntu-latest
+    name: Verify PR Release Note Requirements
+    steps:
+
+      - name: Get Pull Request Number
+        id: get-pull-request-number
+        run: |
+          PULL_REQUEST_NUMBER=$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")
+          echo "PULL_REQUEST_NUMBER=$PULL_REQUEST_NUMBER" >> $GITHUB_OUTPUT
+
+      - name: Check if we have modified release_notes.json 
+        id: get-modified-files
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PULL_NUMBER: ${{steps.get-pull-request-number.outputs.PULL_REQUEST_NUMBER}}
+        run: |
+          FILES_RESPONSE=$(gh api /repos/flutter/devtools/pulls/$PULL_NUMBER/files)
+          echo "FILES_RESPONSE: $FILES_RESPONSE"
+          
+          HAS_CHANGED_RELEASE_NOTES=$(echo $FILES_RESPONSE | jq '.[].filename' | jq -s '. | any(. == env.CURRENT_RELEASE_JSON_FILE_PATH)')
+          echo "HAS_CHANGED_RELEASE_NOTES=$HAS_CHANGED_RELEASE_NOTES" >> $GITHUB_OUTPUT
+
+      - name: Get PR Description
+        id: check-release-note-exceptions
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PULL_NUMBER: ${{steps.get-pull-request-number.outputs.PULL_REQUEST_NUMBER}}
+        run: |
+          PULLS_RESPONSE=$(gh api /repos/flutter/devtools/pulls/$PULL_NUMBER)
+          DESCRIPTION_BODY=$(echo $PULLS_RESPONSE | jq '.body')
+          echo $DESCRIPTION_BODY
+          if $(echo $DESCRIPTION_BODY | grep -q "> NO RELEASE NOTE CHANGES:"); then
```

Comment:
Can we make this shorter? NO_RELEASE_NOTES? And also all one word with underscores so that it is easy to double click copy and paste.

Additionally, we should probably drop the "> " and ":" from the grep so that the gate is a bit wider and someone doesn't end up getting bit by this because they wrote "=" instead of ":" for example.